zerver/actions/message_send.py

```diff
@@ -2169,3 +2170,29 @@ def internal_send_group_direct_message(
 
     sent_message_result = do_send_messages([message])[0]
     return sent_message_result.message_id
+
+
+def send_user_profile_update_notification(
+    user_profile: UserProfile,
+    acting_user: UserProfile | None,
+    changes: list[UserProfileChangeDict],
+) -> None:
+    if acting_user is None or acting_user.id == user_profile.id:
+        return
```

Comment:
I'm not sure whether skipping for `acting_user is None` is correct -- typically that'll be something like the LDAP integration syncing this field. It might be more correct to have some fixed language for that, like "An automated process has made the following changes to your account:"

Or it might be better to skip -- in which case we'd want a comment here explaining the reasoning. This question deserves a #design conversation for more visibilty.

Reply:
OK, it looks like this was specified in the issue and the decision there seems correct, so we can run with how you've done it in the 3rd commit. I think you can squash commits 1, 3, 4, though -- this feature is more readable that way, since half the changes in commits 3 and 4 are just revisions to commit 1, with no structural reason that it'd be helpful to read them that way.

I'd also add a comment on the `acting_user is None` case of the form to explain the reasoning/purpose:
> We can't name which user made the change, since it was done either on the command line or via an automated job like LDAP synchronization. It's still useful to tell the user about the update.